daemon/remote.c

```diff
@@ -2108,6 +2108,7 @@ static int flush(char const* rpcurl, tr_variant** benc)
             break;
 
         default:
+            evbuffer_add_printf(buf, '\0');
```

Comment:
According to the documentation, the signature is:

```c
int evbuffer_add_printf(struct evbuffer *buf, const char *fmt, ...);
```

What this means is you're essentially passing `NULL` (`'\0'` equals `0`) as a format string. Documentation also likens this function to a `printf` C library function.

GCC will give you a warning:

```
example.c: In function ‘main’:
example.c:5:5: warning: null argument where non-null required (argument 1) [-Wnonnull]
     printf('\0');
     ^~~~~~
```

MSDN states that (for `printf`, and for `_vsnprintf` used internally by libevent):

> If `format` is `NULL`, the invalid parameter handler is invoked

I'd add a null byte explicitly rather than rely on current (potentially wrong) behavior, i.e.

```c
evbuffer_add(buf, "", 1);
```